fuel/src/main/kotlin/com/github/kittinunf/fuel/toolbox/HttpClient.kt

```diff
@@ -193,16 +198,20 @@ internal class HttpClient(
         )
     }
 
-    private fun dataStream(connection: HttpURLConnection): InputStream? {
+    private fun dataStream(connection: HttpURLConnection, hook: Hook?): InputStream? {
         return try {
             try {
-                BufferedInputStream(connection.inputStream)
+                val inputStream = hook?.interpretResponseStream(connection.inputStream) ?: connection.inputStream
```

Comment:
Can you make a no-op implementation and remove nullability?

Reply:
(It would pass the argument, so we don't need ` hook?.interpretResponseStream(connection.inputStream) ?: connection.inputStream`, but just ` hook.interpretResponseStream(connection.inputStream)`)